samples/observability.js

```diff
@@ -0,0 +1,284 @@
+  NodeTracerProvider,
+  TraceIdRatioBasedSampler,
+} = require('@opentelemetry/sdk-trace-node');
+const {
+  BatchSpanProcessor,
+  ConsoleSpanExporter,
+  SimpleSpanProcessor,
+} = require('@opentelemetry/sdk-trace-base');
+const {
+  SEMRESATTRS_SERVICE_NAME,
+  SEMRESATTRS_SERVICE_VERSION,
+} = require('@opentelemetry/semantic-conventions');
+
+const resource = Resource.default().merge(
+  new Resource({
+    [SEMRESATTRS_SERVICE_NAME]: 'spanner-sample',
+    [SEMRESATTRS_SERVICE_VERSION]: 'v1.0.0', // The version of your app running.,
+  })
+);
+
+// Create the Google Cloud Trace exporter for OpenTelemetry.
+const {
+  TraceExporter,
+} = require('@google-cloud/opentelemetry-cloud-trace-exporter');
+const exporter = new TraceExporter();
+
+// Optionally, you can enable gRPC instrumentation.
+if (process.env.ENABLE_GRPC_TRACING === 'true') {
+  const {registerInstrumentations} = require('@opentelemetry/instrumentation');
+  const {GrpcInstrumentation} = require('@opentelemetry/instrumentation-grpc');
+  registerInstrumentations({
+    instrumentations: [new GrpcInstrumentation()],
+  });
+}
+
+function traceAndExportSpans(instanceId, databaseId, projectId) {
+  // Wire up the OpenTelemetry SDK instance with the exporter and sampler.
+  const sdk = new NodeSDK({
+    resource: resource,
+    traceExporter: exporter,
+    // Trace every single request to ensure that we generate
+    // enough traffic for proper examination of traces.
+    sampler: new TraceIdRatioBasedSampler(1.0),
+  });
+  sdk.start();
+
+  // Create the tracerProvider that the exporter shall be attached to.
+  const provider = new NodeTracerProvider({resource: resource});
+  provider.addSpanProcessor(new BatchSpanProcessor(exporter));
+
+  // This makes a global tracerProvider but you could optionally
+  // instead pass in the provider while creating the Spanner client.
+  provider.register();
+
+  // Acquire the tracer.
+  const tracer = provider.getTracer('MyApp');
+
+  // Start our user defined trace.
+  tracer.startActiveSpan('deleteAndCreateDatabase', async span => {
+    // Create the Cloud Spanner Client.
+    const {Spanner} = require('@google-cloud/spanner');
+    const spanner = new Spanner({
+      projectId: projectId,
+      observabilityConfig: {
+        // Optional, can rather register the global tracerProvider
+        tracerProvider: provider,
+        enableExtendedTracing: true, // Optional but can also use SPANNER_EXTENDED_TRACING=true
+      },
+    });
+
+    // Acquire the database and databaseAdminClient handles.
+    const instance = spanner.instance(instanceId);
+    const database = instance.database(databaseId);
+    const databaseAdminClient = spanner.getDatabaseAdminClient();
+
+    const databasePath = databaseAdminClient.databasePath(
+      projectId,
+      instanceId,
+      databaseId
+    );
+
+    // Mimicking how in real world code, there will be a pause after
+    // application startup, as the service waits to serve traffic.
```

Comment:
I don't think its required for sample. This is a public sample, so adding too many things is confusing.